stetho/src/main/java/com/facebook/stetho/inspector/elements/android/MethodInvoker.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.elements.android;
+
+import com.facebook.stetho.common.LogUtil;
+import com.facebook.stetho.common.Util;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Tries to arbitrarily invoke single argument methods by name on an object instance by trying out
+ * different argument types.
+ */
+public class MethodInvoker {
+
+  private static final List<TypedMethodInvoker<?>> invokers = Arrays.asList(
+      new StringMethodInvoker(), new CharSequenceMethodInvoker(), new IntegerMethodInvoker(),
+      new FloatMethodInvoker(), new BooleanMethodInvoker());
+
+  /**
+   * Tries to invoke a method on receiver with a single argument by trying out different types
+   * for arg until it finds one that matches (or not). No exceptions are thrown on failure.
+   *
+   * @param methodName The method name to be invoked
+   * @param argument The single argument to be provided to the method
+   */
+  public void invoke(Object receiver, String methodName, String argument) {
+    Util.throwIfNull(receiver, methodName, argument);
+    for (TypedMethodInvoker<?> invoker : invokers) {
+      if (invoker.invoke(receiver, methodName, argument)) {
+        return;
+      }
+    }
+  }
+
+  private static abstract class TypedMethodInvoker<T> {
+    private Class<T> mArgType;
+
+    TypedMethodInvoker(Class<T> argType) {
+      mArgType = argType;
+    }
+
+    boolean invoke(Object receiver, String methodName, String argument) {
+      try {
+        Method method = receiver.getClass().getMethod(methodName, mArgType);
+        method.invoke(receiver, convertArgument(argument));
+        return true;
+      } catch (NoSuchMethodException e) {
+        LogUtil.w("Method with name " + methodName + " and argument of type " +
```

Comment:
Won't this create a lot of debug spew? For instance, something like `setAlpha` is always going to fail with String, then fail with Char, then fail with Integer, before finally succeeding with Float. We'll get 3 log outputs with that every time, right? Probably better to only log in the calling method if it can't find any `invoke()` that returns true. The `InvocationTargetException`s should be fine though.